### PR TITLE
Implement update passes

### PR DESCRIPTION
**Pull request by PoignardAzur**
Implement update_anim pass.
Implement update_focus pass.

### CODE REVIEW

File: masonry/src/widget/widget_state.rs [hunk 1]
```diff
@@ -90,6 +90,7 @@ pub struct WidgetState {
     pub(crate) needs_accessibility_update: bool,
 
     /// Any descendant has requested an animation frame.
+    pub(crate) needs_anim: bool,
     pub(crate) request_anim: bool,
```
Comment (jaredoconnell) [inline masonry/src/widget/widget_state.rs:94]: I think some differentiating comments for these two fields would be helpful. Looking at the code, it looks like needs_anim allows it to start the update anim pass, and request_anim instructs it to send the lifecycle event during the update anim pass. If so, I think clearer variable names could be helpful.

Comment (PoignardAzur) [reply masonry/src/widget/widget_state.rs:94]: It's not clearly documented yet, but a naming scheme I'm starting to adopt for these flags is that "request_x" means "this widget requested x" and "needs_x" means "this widget or a descendant requested x".

File: masonry/src/passes/update.rs [hunk 2]
```diff
@@ -145,3 +146,86 @@ pub(crate) fn run_update_pointer_pass(root: &mut RenderRoot, root_state: &mut Wi
     // Pass root widget state to synthetic state create at beginning of pass
     root_state.merge_up(root.widget_arena.get_state_mut(root.root.id()).item);
 }
+
+// ----------------
+
+fn recurse_on_children(
```
Comment (DJMcNab) [inline masonry/src/passes/update.rs:152]: This looks extremely similar to the function in `passes/compose.rs`. Please document the differences here

Comment (PoignardAzur) [reply masonry/src/passes/update.rs:152]: It's the same. I'll remove the duplicate once #522 is merged.

File: masonry/src/widget/widget_state.rs (hunk 1 above)
Comment (PoignardAzur) [reply masonry/src/widget/widget_state.rs:94]: I've added some comments in #522. Will add more here later.

Comment (PoignardAzur) [reply masonry/src/widget/widget_state.rs]: Done.

File: masonry/src/passes/update.rs (hunk 2 above)
Comment (PoignardAzur) [reply masonry/src/passes/update.rs]: Done.

**Comment by PoignardAzur** [comment]
I'm closing this PR. Will reopen as two separate PRs.

**Pull request closed without merging**